code/modules/mob/observer/ghost/ghost.dm

```diff
@@ -61,11 +61,15 @@ var/global/list/image/ghost_sightless_images = list() //this is a list of images
 				mind = new /datum/mind(key)
 				mind.current = src
 	if(!T)
-		var/list/spawn_locs = global.latejoin_locations | global.latejoin_cryo_locations | global.latejoin_gateway_locations
-		if(length(spawn_locs))
-			T = pick(spawn_locs)
-		else
-			T = locate(1, 1, 1)
+
+		var/list/spawn_locs = list()
+		var/list/all_spawns = decls_repository.get_decls_of_subtype(/decl/spawnpoint)
+		for(var/spawn_type in all_spawns)
+			var/decl/spawnpoint/spawn_data = all_spawns[spawn_type]
+			if(spawn_data.ghost_can_spawn && length(spawn_data.spawn_turfs))
+				spawn_locs |= spawn_data.spawn_turfs
+		T = length(spawn_locs) ? pick(spawn_locs) : get_respawn_loc()
```

Comment:
Also I feel as though it would be better to use get_respawn_loc() and have that proc do checks on either flags or using another proc that takes the mob as a parameter.